packages/core/src/lib/__tests__/modal.spec.ts

```diff
@@ -1,38 +1,71 @@
+import { render, fireEvent, screen, within } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
 import Modal from '../modal.svelte';
+import { writable } from 'svelte/store';
 
 describe('Modal', () => {
-  it('should emit close event when close icon button is clicked', async () => {
-    const { getByTitle, component } = render(Modal, {
-      open: true,
+  let isOpen = writable(true);
+  let currentValue = true;
+
+  beforeEach(() => {
+    isOpen = writable(true);
+    currentValue = true;
+
+    const unsubscribe = isOpen.subscribe((value) => {
+      currentValue = value;
+    });
+
+    afterEach(() => {
+      unsubscribe();
     });
+  });
+
+  it('should close modal when close icon button is clicked', async () => {
+    render(Modal, { isOpen });
 
-    const closeButton = getByTitle('Close modal');
-    const onClose = vi.fn();
-    component.$on('close', onClose);
-    await fireEvent.click(closeButton);
+    const modal = screen.getByRole('dialog');
+    const closeButton = within(modal).getByRole('button', { name: /close/iu });
 
-    expect(onClose).toBeCalledWith(expect.objectContaining({ detail: true }));
+    await userEvent.click(closeButton);
+
+    expect(currentValue).toBe(false);
   });
 
-  it('should emit close event when clicked outside the modal', async () => {
-    const { container, component } = render(Modal, {
-      open: true,
-    });
+  it('should close modal when clicked outside the modal', async () => {
+    render(Modal, { isOpen });
 
-    const onClose = vi.fn();
-    component.$on('close', onClose);
+    const modal = screen.getByRole('dialog');
+    await userEvent.click(modal.parentElement!);
 
-    const background = container.querySelector('div[role="button"]');
-    if (!background) {
-      throw new Error('Background not found');
-    }
+    expect(currentValue).toBe(false);
+  });
+
+  it('if open is true, modal should be visible', () => {
+    render(Modal, { isOpen });
+    const modal = screen.queryByRole('dialog');
+    expect(modal).toBeInTheDocument();
+    expect(modal).toHaveAttribute('aria-modal', 'true');
+  });
 
-    await fireEvent.click(background);
+  it('if open is false, modal should not be visible', () => {
+    isOpen.set(false);
+    render(Modal, { isOpen });
+    const modal = screen.queryByRole('dialog');
+    expect(modal).not.toBeInTheDocument();
+  });
+
+  it('should close modal when escape key is pressed', async () => {
+    render(Modal, { isOpen });
+
+    await fireEvent.keyDown(window, { key: 'Escape' });
+    expect(currentValue).toBe(false);
+  });
 
-    expect(onClose).toHaveBeenLastCalledWith(
-      expect.objectContaining({ detail: true })
-    );
+  it('should focus on heading element on mount', () => {
+    render(Modal, { isOpen });
+    const heading = document.querySelector('#modal-heading');
+    const elementInFocus = document.activeElement;
+    expect(elementInFocus).toEqual(heading);
```

Comment:
- Don't use `querySelector`, use `getByRole`
- Unless it's semantically important (e.g. it's the ID of a label element that is passed to `aria-labeledby`) IDs should not be used as test hooks
- Use [`expect(...).toHaveFocus()`](https://github.com/testing-library/jest-dom#tohavefocus) to check focus

```suggestion
    render(Modal, { isOpen });
    
    const modal = screen.getByRole('dialog');
    const heading = within(modal).getByRole('heading')

    expect(heading).toHaveFocus()
```